### PR TITLE
Laravel 8 support - bring guzzle adapter in house

### PR DESCRIPTION
**Pull request by edalzell**
The Guzzle Adapter package doesn't support Guzzle 7 via constraints but does work with no changes. License was MIT, so brought in house so we can support Laravel 8.

All tests pass.

### CODE REVIEW

**Comment by edalzell** [comment]
Alrighty all tests run locally anyway.

**Comment by damcclean** [comment]
I think the reasons the tests are failing is due to the `laravel/legacy-factories` package requiring a minimum of PHP 7.3 but the tests are running for PHP 7.2 (as that's what Statamic supports). 

I would have thought moving it to be a dev dependency would have sorted but doesn't seem like it. 🤔

**Comment by jasonvarga** [comment]
We're only using a model factory in one file:

https://github.com/statamic/cms/blob/master/tests/Auth/Eloquent/EloquentUserTest.php#L101
https://github.com/statamic/cms/blob/master/tests/Auth/Eloquent/EloquentUserTest.php#L34-L41

We can probably just avoid using a model factory there and then get rid of the dependency.

**Comment by edalzell** [comment]
Having some difficulty w/ fake user generation and non-unique `id`s

**Comment by mrmonat** [comment]
I have opened a PR on your fork which fixes all the failing tests by not using laravel/legacy-factories but instead using Faker on the failing test directly. 

PR: https://github.com/edalzell/cms/pull/415

**Comment by edalzell** [comment]
@jasonvarga I tried to get the tests running properly but GH actions befuddle me. I _think_ I have it set up so that L8 won't run w/ PHP 7.2, but now L8 isn't running at all.

**Comment by mrmonat** [comment]
You have merged my PR only "half". You use "WithFaker" Trait in the Test now without using it.

File: tests/Auth/Eloquent/EloquentUserTest.php [hunk 1]
```diff
@@ -17,7 +16,9 @@
 
 class EloquentUserTest extends TestCase
 {
-    use UserContractTests, PermissibleContractTests, HasPreferencesTests;
+    use UserContractTests, PermissibleContractTests, HasPreferencesTests, WithFaker;
+
+    private $num = 0;
```
Comment (jasonvarga) [inline tests/Auth/Eloquent/EloquentUserTest.php:21]: See my other comment. This property can be removed.

**Review by jasonvarga — CHANGES_REQUESTED** [review]

**Comment by jasonvarga** [comment]
Thanks very much for this. 🎉  @mrmonat, too.

**Pull request merged**